### ISSUE TITLE
Avoid inline Javascript in Admin main template

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#6354]
# Problem description

Currently the admin main template is setting the SULU_CONFIG via a inline javascript file: https://github.com/sulu/sulu/blob/2.3.6/src/Sulu/Bundle/AdminBundle/Resources/views/Admin/main.html.twig

# Proposed solution

Instead of setting this configuration VIA javascript the startAdmin component could read this configuration from the DOM itself e.g.:

```twig
<div id="application" data-config="{{ config|json_encode }}"></div>
```

This will make it in future possible to use a strict CSP headers for the admin when [lodash issue](https://github.com/lodash/lodash/issues/832) is solved.

**Comment by alexander-schranz** [comment]
To achieve this also the `startAdmin` need to be adjusted, which will need to set the `SULU_CONFIG` here for bc reasons: https://github.com/sulu/sulu/blob/0ea88d19c560cde4a947c76f4a4477f2f43b6c8b/src/Sulu/Bundle/AdminBundle/Resources/js/index.js#L354

**Comment by vvasiloi** [comment]
@alexander-schranz How do you suggest keeping the BC for the `config` block?
https://github.com/sulu/sulu/blob/6e3451bbd04423fdb9c7b63161ff9aeafa1bdbdf/src/Sulu/Bundle/AdminBundle/Resources/views/Admin/main.html.twig#L29-L31

**Comment by alexander-schranz** [comment]
@vvasiloi I even don't think that somebody overwrite it but we can just add a block config around it still:

```twig
<div id="application" data-config="{%- block config -%}{{- config|json_encode -}}{%- endblock -%}"></div>
```

This way the only usecase to overwrite the config block which I think still works:

```twig
{% block config %}
    {% set config = config|merge({custom: 'value'}) %}
    {{ parent() }}
{% endblock %}
```

**Comment by vvasiloi** [comment]
@alexander-schranz should it be deprecated?

**Comment by vvasiloi** [comment]
Also, I don't know how to approach the JS part of it. The config is currently stored in a constant but is retrieved through the `Config` wrapper. I understand that you want the constant deprecated and kept for BC only. How do you propose to keep the new state?

**Comment by alexander-schranz** [comment]
As written in https://github.com/sulu/sulu/issues/6354#issuecomment-1081679586 the `SULU_CONFIG` variable creation will be moved to `startAdmin` method. We can achieve it via `window.SULU_CONFIG = ...` so it is still a global.

**Comment by vvasiloi** [comment]
So there won't be a new way to define the configuration? It's still going to be the `SULU_CONFIG` constant, just defined in a different place?

**Comment by alexander-schranz** [comment]
@vvasiloi Yes, correctly. The part here is get ride of the inline javascript in the admin html twig, so in future we can have stricter and securer CSP (Content Security Policy). There are other issues open but this is atleast one required part of it :)

**Comment by vvasiloi** [comment]
@alexander-schranz alright, I think I got it, thanks! Working on the PR.

**Comment by vvasiloi** [comment]
> As written in [#6354 (comment)](https://github.com/sulu/sulu/issues/6354#issuecomment-1081679586) the `SULU_CONFIG` variable creation will be moved to `startAdmin` method. We can achieve it via `window.SULU_CONFIG = ...` so it is still a global.

@alexander-schranz  this doesn't work, because the constant doesn't exist when it's exported in the Config service:
https://github.com/sulu/sulu/blob/6e3451bbd04423fdb9c7b63161ff9aeafa1bdbdf/src/Sulu/Bundle/AdminBundle/Resources/js/services/Config/Config.js#L4

**Comment by alexander-schranz** [comment]
@vvasiloi We have here 2 options. **A:** Read the config inside `Config.js` from the DOM  instead of startAdmin or **B:** do a `export default {};` inside Config.js and a `Object.assign(Config, JSON.parse(jsonConfig));` in the startAdmin.

**Comment by alexander-schranz** [comment]
Fixed in #6964
